railties/lib/rails/commands/temp/temp_command.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require "rails/commands/temp/helpers"
+
+module Rails
+  module Command
+    class TempCommand < Base # :nodoc:
+      include Rails::Command::Temp::Helpers
+      namespace "tmp"
+      desc :clear, "Clear cache, socket and screenshot files from tmp/ (narrow w/ tmp:cache:clear, tmp:sockets:clear, tmp:screenshots:clear)"
+      def clear
+        Rails::Command.invoke "tmp:cache:clear"
+        Rails::Command.invoke "tmp:sockets:clear"
+        Rails::Command.invoke "tmp:screenshots:clear"
+        Rails::Command.invoke "tmp:storage:clear"
+      end
+
+      # desc :create, "Create tmp directories for cache, sockets, and pids"
+      def create
+        create_tmp_dir "cache"
+        create_tmp_dir "sockets"
+        create_tmp_dir "pids"
+        create_tmp_dir "cache/assets"
+      end
+    end
+
+    module Temp
+      class CacheCommand < Base # :nodoc:
```

Comment:
Unfortunately, putting this module in `railties/lib/rails/commands/temp/temp_command.rb` means [`lookup`](https://github.com/rails/rails/blob/e23e1718ead861ae43004f5706ddad780a440359/railties/lib/rails/command/behavior.rb#L36-L53) won't find these commands (because [`namespaces_to_paths`](https://github.com/rails/rails/blob/e23e1718ead861ae43004f5706ddad780a440359/railties/lib/rails/command/behavior.rb#L70-L80) won't return the right path):

```console
$ bin/rails tmp:screenshots:clear
Unrecognized command "tmp:screenshots:clear" (Rails::Command::UnrecognizedCommandError)
Did you mean?  tmp:sockets:clear

$ bin/rails tmp:sockets:clear
Unrecognized command "tmp:sockets:clear" (Rails::Command::UnrecognizedCommandError)
Did you mean?  tmp:cache:clear

$ bin/rails tmp:cache:clear
Unrecognized command "tmp:cache:clear" (Rails::Command::UnrecognizedCommandError)
Did you mean?  tmp:clear
```

(But "Did you mean?" can subsequently suggest them because [it uses `printing_commands`](https://github.com/rails/rails/blob/e23e1718ead861ae43004f5706ddad780a440359/railties/lib/rails/command.rb#L43).)


So each command needs to go into a dedicated subdirectory.

Reply:
I've extracted these to separate files.